webui/src/components/Scan.vue

```diff
@@ -227,6 +228,20 @@ export default {
       });
     },
 
+    deletePreview() {
+      this.$emit('mask', 1);
+      Common.fetch('preview', {
+        method: 'DELETE'
+      }).then(data => {
+        this.$emit('notify', { type: 'i', message: `Deleted preview` });
+        window.setTimeout(this.readPreview, 1000);
```

Comment:
Since the deletion has now occurred, can you just call `this.readPreview()` directly rather than using `window.setTimeout`? (My hunch is yes, but you may have run into issues). I prefer to avoid `window.setTimeout` unless it's essential. If it needs to happen out of this cycle then use [nextTick](https://vuejs.org/v2/api/#Vue-nextTick)